src/core/manager.c

```diff
@@ -3934,6 +3936,19 @@ static int build_generator_environment(Manager *m, char ***ret) {
                         return r;
         }
 
+        cv = detect_confidential_virtualization();
+        if (cv < 0)
+                log_debug_errno(cv, "Failed to detect confidential virtualization, ignoring: %m");
+        else if (v > 0) {
+                const char *s;
+
+                s = strjoina(confidential_virtualization_to_string(cv));
+
+                r = strv_env_assign(&nl, "SYSTEMD_CONFIDENTIAL_VIRTUALIZATION", s);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
if you add this here (which is ok), then please also add brief docs to bottom of systemd.generator(7)